src/rpc_client/mod.rs

```diff
@@ -91,14 +92,17 @@ impl ApiInfo {
         ApiInfo::from_str(&api_info)
     }
 
-    pub async fn call<T: HasLotusJson>(&self, req: RpcRequest<T>) -> Result<T, JsonRpcError> {
+    pub async fn call<T: HasLotusJson + std::fmt::Debug>(
+        &self,
+        req: RpcRequest<T>,
+    ) -> Result<T, JsonRpcError> {
         let rpc_req = RequestObject::request()
             .with_method(req.method_name)
             .with_params(req.params)
             .with_id(0)
             .finish();
 
-        let api_url = multiaddress_to_url(&self.multiaddr, req.rpc_endpoint).to_string();
+        let api_url = multiaddress_to_url(&self.multiaddr, req.rpc_endpoint, false).to_string();
```

Comment:
How about using an enum to improve readability here, e.g. `multiaddress_to_url(&self.multiaddr, req.rpc_endpoint, Protocol::WS)`